src/Speech/SpeechClient.php

```diff
@@ -0,0 +1,494 @@
+     *           requests specifically for authentication.
+     *     @type callable $httpHandler A handler used to deliver Psr7 requests.
+     *     @type string $keyFile The contents of the service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type int $retries Number of retries for a failed request. Defaults
+     *           to 3.
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     * @throws \InvalidArgumentException
+     */
+    public function __construct(array $config = [])
+    {
+        if (!isset($config['scopes'])) {
+            $config['scopes'] = [self::SCOPE];
+        }
+
+        $this->connection = new Rest($this->configureAuthentication($config));
+    }
+
+    /**
+     * Runs a recognize request and returns the results immediately. Ideal when
+     * working with audio up to approximately one minute in length.
+     *
+     * The Google Cloud Client Library will attempt to infer the sample rate
+     * and encoding used by the provided audio file for you. This feature is
+     * recommended only if you are unsure of what the values may be and is
+     * currently limited to .flac, .amr, and .awb file types. The sample rate
+     * cannot be inferred from audio provided from a Google Storage object.
+     *
+     * Example:
+     * ```
+     * $results = $speech->recognize(
+     *     fopen('audio.flac', 'r')
+     * );
+     *
+     * foreach ($results as $result) {
+     *     echo $result['transcript'];
+     * }
+     * ```
+     *
+     * ```
+     * // Run with speech context, sample rate, and encoding provided
+     * $results = $speech->recognize(
+     *     fopen('audio.flac', 'r'), [
+     *     'encoding' => 'FLAC',
+     *     'sampleRate' => 16000,
+     *     'speechContext' => [
+     *         'phrases' => [
+     *             'The Google Cloud Platform',
+     *             'Speech API'
+     *         ]
+     *     ]
+     * ]);
+     *
+     * foreach ($results as $result) {
+     *     echo $result['transcript'];
+     * }
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/speech/reference/rest/v1beta1/speech/syncrecognize#SpeechRecognitionAlternative SpeechRecognitionAlternative
+     * @see https://cloud.google.com/speech/reference/rest/v1beta1/speech/syncrecognize SyncRecognize API documentation
+     * @see https://cloud.google.com/speech/reference/rest/v1beta1/RecognitionConfig#AudioEncoding AudioEncoding types
+     * @see https://cloud.google.com/speech/docs/best-practices Speech API best practices
+     * @codingStandardsIgnoreEnd
+     *
+     * @param resource|string|Object $audio The audio to recognize. May be a
```

Comment:
Yes, that sounds like a good plan. Let's open a separate PR after the fact that addresses any methods that accept this type of input and make sure we hit them all at the same time.